core/src/main/java/hudson/slaves/CloudRetentionStrategy.java

```diff
@@ -44,36 +45,49 @@
  */
 public class CloudRetentionStrategy extends RetentionStrategy<AbstractCloudComputer> {
     private int idleMinutes;
+    private transient ReentrantLock checkLock;
 
     public CloudRetentionStrategy(int idleMinutes) {
         this.idleMinutes = idleMinutes;
+        readResolve();
```

Comment:
Why do you need call readResolve() from constructor? 
readResolve() is called automatically during xml deserialisation and usually should check/update old configuration to new.